synapse/handlers/federation.py

```diff
@@ -263,6 +264,7 @@ async def on_receive_pdu(
         state = None
 
         # Get missing pdus if necessary.
+        # We don't need to worry about outliers because TODO!
```

Comment:
Unrelated to the PR but just want to better clarify why we skip outliers in this step. Any insight here?

 - Outliers are outside of the normal DAG
 - We don't have state for outliers
 - We trust their auth events over calculating them

Reply:
This bit of code is basically implementing the additional checks on events that are part of the "timeline" (on top of the checks we'll do anyway for outliers). Specifically: a) fetching any missing prev events to fill in gaps in the graph, and b) fetching state if we have a hole in the graph.

> Outliers are outside of the normal DAG

More accurately, they're outside the chunk of the DAG we currently have. 

> We trust their auth events over calculating them

There are two level of checks:

1. Check the event passes auth based on `auth_events`, this allows us to assert that the event would have been allowed at some point. If an event passes this check its OK for it to be used as part of a returned `/state` request, as either a) we received the event as part of the original join and so trust it, or b) we'll do a state resolution with existing state before it becomes part of the "current state", which adds more protection.
2. Check the event passes auth based on the state at the event. This is done for events that are to be added to the timeline.

Reply:
Added this info to a comment there 🤔

 1. For 1, I think this is in `_process_received_pdu`
 1. For 2, I think this is the current position of this comment in `on_receive_pdu`